### ISSUE TITLE
`no-this-in-template-only-components` is triggered in class backed gts components

### ISSUE THREAD

**Issue by johanrd** [ember-template-lint/ember-template-lint#3217]
### Steps to reproduce:
1. Have a class backed component in template tag format:

```glimmer-ts
import { on } from '@ember/modifier';
import Component from '@glimmer/component';

export default class Example extends Component {

  handleClick = (event: Event) => {
    console.log('Button clicked!', event);
  };

  <template>
    <button {{on 'click' this.handleClick}} type='button'>+</button>
  </template>
}
```

2. Enable `no-this-in-template-only-components` in `.template-lintrc.js` (as this is proposed as default true in https://github.com/ember-template-lint/ember-template-lint/issues/3210)

```diff
module.exports = {
  extends: ["recommended"],
  rules: {
+    'no-this-in-template-only-components': true
  },
}
```

3. Experience unexpected error when running `pnpm glint`:
```
app/components/a.gts
  11:25  error  Usage of 'this' in path 'this.handleClick' is not allowed in a template-only component. Use '@handleClick' if it is a named argument or create a component.js for this component.  no-this-in-template-only-components
```

### Expected behavior:
The rule `no-this-in-template-only-components` should not be triggered from class backed components

**Comment by lifeart** [comment]
Hi @johanrd, thank yo for report, issue should be fixed here - https://github.com/ember-template-lint/ember-template-lint/pull/3240
